Grinnell-Events-iOS/Grinnell-Events-iOS/ViewController.m

```diff
@@ -254,22 +249,25 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
 
 - (void)scrollViewDidScroll:(UIScrollView *)scrollView
 {
-    
     NSArray *visibleRows = [self.tableView visibleCells];
     UITableViewCell *firstVisibleCell = [visibleRows objectAtIndex:0];
     NSIndexPath *path = [self.tableView indexPathForCell:firstVisibleCell];
     
-    
     //Scroll to the selected date.
     NSDate *toDate = [NSDate dateFromString:self.sortedDateKeys[path.section] ];
     
-    NSDateComponents *firstComponents = [[NSCalendar currentCalendar] components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear fromDate:toDate];
     
-    NSInteger year = [firstComponents year];
-    NSInteger month = [firstComponents month];
-    NSInteger day = [firstComponents day];
+    if (![toDate isEqualToDate:self.currentDateViewed]){
+        self.currentDateViewed = toDate;
+        NSDateComponents *firstComponents = [[NSCalendar currentCalendar] components:NSCalendarUnitDay | NSCalendarUnitMonth | NSCalendarUnitYear fromDate:self.currentDateViewed];
+    
+        NSInteger year = [firstComponents year];
+        NSInteger month = [firstComponents month];
+        NSInteger day = [firstComponents day];
+    
+        [self.dayPicker setCurrentDate:[NSDate dateFromDay:day+1 month:month year:year] animated:YES];
```

Comment:
Break this into two lines:
```
NSDate *followingDay = [NSDate dateFromDay:day+1 month:month year:year];
[dayPicker setCurrentDate:followingDay animated:YES];
```
Use a good variable name so your method calls sound more like sentences when you read them out loud.